File: quicksort.py
def quicksort(arr,left,right):
	i = left; j = right;
	pivot= (left+right)/2;
	while(i<=j):
		while(arr[i]<arr[pivot]):
			i=i+1
		while(arr[j]>arr[pivot]):
			j=j+1
		if(i<=j):
			tmp = arr[i]
			arr[i] = arr[j]
			arr[j] = tmp	
			i= i+1
			j = j-1

	if(left<j):
		return quicksort(arr,left,j)
	if(i<right):
		return quicksort(arr,i,right)

arr = [1,90,2,4,-5,-20,-7,-89,80,2,0]
print arr
quicksort(arr,0,10)
print arr

File: quicksort.cpp


#include <iostream>
using namespace std;



void quickSort(int arr[], int left, int right){
	int i = left, j = right;
	int tmp;
	int pivot = arr[(left+right)/2];

	//partition

	while(i<=j){
		while(arr[i]<pivot){
			i++;
		}
		while(arr[j]>pivot){
			j--;
		}
		if(i<=j){
			tmp = arr[i];
			arr[i] = arr[j];
			arr[j] = tmp;;
			i++;
			j--;
		}
	}

	//recursion
	if (left<j)
		quickSort(arr,left,j);
	if (i<right)
		quickSort(arr,i,right);

}

void printarr(int arr[], int len){
	for(int i=0; i<len; i++){
		cout << arr[i] << endl;
	}
} 



int main(){
	cout << "testing quicksort in cpp" << endl;
	int arr[10] = { 28, 2, 9, 10, 11, 0,-5, 9, 2, 1 };
	int len = sizeof(arr)/sizeof(arr[0]); 
	printarr(arr,len);
	quickSort(arr, 0,len-1);
	cout << endl;
	printarr(arr,len);
}
